File: WolfDen/Game/Behaviours/TransportBehaviour/TransportableBehaviour.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using WolfDen.Game;
using WolfDen.Game.Interfaces;

namespace WolfDen.Game.Behaviours.TransportBehaviour
{
    public class TransportableBehaviour : IBehaviour
    {
        public string Id { get; set; }
        public string Class { get { return GetType().Name; } }
        public Unit Unit { get; }
        public Game Game { get; }
        public IMoveBehaviour MoveBehaviour { get; }

        public TransportableBehaviour(Unit unit, Game game, IMoveBehaviour moveBehaviour)
        {
            Unit = unit;
            Game = game;
            MoveBehaviour = moveBehaviour;
        }

        public bool ExecuteMove(dynamic data)
        {
            string transportBehaviourUnitIdString = data.unitId;
            string transportBehaviourUnitId = transportBehaviourUnitIdString;
            Unit transportBehaviourUnit = (Unit)Game.Pieces[transportBehaviourUnitId];
            TransportBehaviour transportBehaviour = transportBehaviourUnit.Behaviours.Values.OfType<TransportBehaviour>().FirstOrDefault();
            return transportBehaviour.LoadUnit(Unit);
        }

        public void Destroy()
        {
        }
    }
}

File: WolfDen/wwwroot/ts/Behaviours/TransportBehaviour/TransportableBehaviour.ts
namespace TurnBasedBoardGame {
    export class TransportableBehaviour implements IExecuteMove, IDestroy {
        Unit: Unit;
        MoveBehaviour: IMoveBehaviour;
        Id: string;
        Game: Game;

        Initialize() {
            let self = this;
            this.Unit.ActiveClickEvent[this.Id] = function (e) {
                var reachableTiles = self.MoveBehaviour.GetReachableTilesAndPaths();
                reachableTiles.reachableNodes.forEach((tile: Tile) => {
                    if (tile.Piece != null && tile.Piece.Owner == self.Unit.Owner) {
                        let unit = tile.Piece as Unit;
                        let transportBehaviour = unit.BehaviourList.find(IsITransportBehaviour) as ITransportBehaviour;
                        if (transportBehaviour && !transportBehaviour.TransportedUnit) {
                            self.Game.Indicators.push(new LoadTransportIndicator(self.Unit, self, transportBehaviour, tile, reachableTiles.GetIdPath(tile)));
                        }
                    }

                });
            };
        }


        Destroy(): void {
        }

        ExecuteMove(move: any): boolean {
            var self = this;
            let transportBehaviourUnit = Game.Instance.Pieces[move.unitId] as Unit;
            let transportBehaviour = transportBehaviourUnit.BehaviourList.find(IsITransportBehaviour) as ITransportBehaviour;
            let animation = new PathAnimation(move.path, self.Unit, null);
            return transportBehaviour.LoadUnit(self.Unit);
              
        }
    }
}